.github/workflows/build-test-release.yml

```diff
@@ -13,8 +13,54 @@ on:
                 type: string
                 required: true
 
+env:
+    NODE_VERSION: "16.x"
+
 jobs:
-  dummy:
+    build:
         runs-on: ubuntu-latest
         steps:
-          - run: echo placeholder
+            - name: Resolve ref
+              run: |
+                ref="${{github.ref}}"
+                tag="${{github.event.inputs.tag_override}}"
+                [ -n "$tag" ] && ref="refs/tags/$tag"
+                echo "REF=$ref" >> $GITHUB_ENV
+            - name: Fail if not running from a tag
+              if: "!startsWith(env.REF , 'refs/tags/')"
+              run: exit 1
+            - name: Parse tag into env vars
+              # Skip 'refs/tags/' (10 chars), read until '/' ("package"), then read the rest ("version").
+              run: |
+                  echo "${{env.REF}}" |
+                    (read -n 10; read -d/ p; read v; echo "PACKAGE=$p"; echo "VERSION=$v") >> $GITHUB_ENV
+#            - name: debug
+#              run: echo "PACKAGE=$PACKAGE, VERSION=$VERSION"
+            - name: Checkout
+              uses: actions/checkout@v3
+              with:
+                ref: "${{env.REF}}"
+            - name: Check that version matches 'package.json' up to build version
+              working-directory: "./packages/${{env.PACKAGE}}"
+              run: |
+                  v="$(npm pkg get version | jq -r ".[\"@concordium/${{env.PACKAGE}}\"]")"
+                  [[ "${{env.VERSION}}" = ${v}-* ]]
+            - name: Setup Node
+              uses: actions/setup-node@v3
+              with:
+                  node-version: "${{env.NODE_VERSION}}"
+            - name: Install dependencies
+              run: yarn install --immutable --check-cache
+            - name: Build and archive
+              working-directory: "./packages/${{env.PACKAGE}}"
+              run: |
+                yarn build
+                zip dist.zip ./dist/*
+            - name: Upload package as GitHub release
+              uses: softprops/action-gh-release@v1
+              with:
+                  tag_name: "${{env.REF}}"
+                  body: "${{github.event.inputs.release_notes}}"
+                  files: "./packages/${{env.PACKAGE}}/dist.zip"
+                  name: "${{env.PACKAGE}}: ${{env.VERSION}}"
+                  generate_release_notes: false # automatic mechanism doesn't limit changes to the target project
```

Comment:
I don't think I understand this comment. :thinking: 
(Are you saying that we don't generate release notes because it can't generate ones that only contain changes for the target package?)

Reply:
yes you understand that correctly :)